### PR TITLE
Add in-table tools for simple csv generation. Add json column for explicit object->json conversion for the csv

### PR DESCRIPTION
**Pull request by KnewOne**

### CODE REVIEW

File: lib/src/main/java/adminlte/entity_list_table/communication/http/tables/AbstractTable.java [hunk 1]
```diff
@@ -46,4 +44,55 @@ protected void addColumn(ColumnDefinitionInterface column) {
     public String getPaginationParameter() { return this.paginationParameter; }
 
     public String getSearchParameter() { return this.searchParameter; }
+
+    public List<ColumnDefinitionInterface> getCsvColumns() { return new ArrayList<ColumnDefinitionInterface>(); }
```
Comment (lehadnk) [inline lib/src/main/java/adminlte/entity_list_table/communication/http/tables/AbstractTable.java:48]: Honestly, I don't think that it makes a good sense to have a separate method for getCsvColumns. Why don't we use columns defined in the constructor/defineColumns method?

File: lib/src/main/java/adminlte/entity_list_table/communication/http/tables/AbstractTable.java [hunk 2]
```diff
@@ -46,4 +44,55 @@ protected void addColumn(ColumnDefinitionInterface column) {
...
+
+    public String generateCsv(String fileName, boolean addQuotation) {
```
Comment (lehadnk) [inline lib/src/main/java/adminlte/entity_list_table/communication/http/tables/AbstractTable.java:50]: This is bad. Formatting CSV is not part of the AbstractTable class responsibility. I think that we should work with the service's renderTable and TableRender class and provide another layer of abstraction (like TableHtmlRenderer / TableCsvRenderer), or maybe even create a new service to handle CSV exports.

File: lib/src/main/java/adminlte/entity_list_table/communication/http/tables/columns/AbstractColumn.java [hunk 3]
```diff
@@ -24,7 +23,15 @@ public String getFieldName() {
         return this.fieldName;
     }
 
+    public String getCsvCellContent(Object object) {
```
Comment (lehadnk) [inline lib/src/main/java/adminlte/entity_list_table/communication/http/tables/columns/AbstractColumn.java:26]: Again, I'm not sure if it's good behavior. Why don't we use getObjectValue.toString()?

File: lib/src/main/java/adminlte/entity_list_table/communication/http/tables/columns/DateTimeColumn.java [hunk 4]
```diff
@@ -63,23 +62,15 @@ private String getFormattedUtcDateTimeString(Object object) {
...
-        if (content instanceof Date) {
-            return ((Date) content).toLocalDate().atStartOfDay().toInstant(ZoneOffset.UTC);
-        }
+        if (content instanceof Date) { return ((Date)content).toLocalDate().atStartOfDay().toInstant(ZoneOffset.UTC); }
 
         // assuming that LocalDateTime stores UTC value
-        if (content instanceof LocalDateTime) {
-            return ((LocalDateTime) content).toInstant(ZoneOffset.UTC);
-        }
+        if (content instanceof LocalDateTime) { return ((LocalDateTime)content).toInstant(ZoneOffset.UTC); }
```
Comment (lehadnk) [inline lib/src/main/java/adminlte/entity_list_table/communication/http/tables/columns/DateTimeColumn.java:72]: I think that auto formatting did something bad with code readability here.

File: lib/src/main/java/adminlte/entity_list_table/communication/http/tables/AbstractTable.java (hunk 1 above)
Comment (IKnewOne) [reply lib/src/main/java/adminlte/entity_list_table/communication/http/tables/AbstractTable.java:48]: We might not want to have the same info in csv as we do in the table on the page. Flexibility doesn't hurt. And if one needs, he can easily just make the return of this function the defined columns ones.

File: lib/src/main/java/adminlte/entity_list_table/communication/http/tables/columns/AbstractColumn.java (hunk 3 above)
Comment (IKnewOne) [reply lib/src/main/java/adminlte/entity_list_table/communication/http/tables/columns/AbstractColumn.java:26]: Same reasoning, it's more flexibility. For example, the table timestamp values might be processed to show localized times, while csv exports with the timestamp as written in the object

File: lib/src/main/java/adminlte/entity_list_table/communication/http/tables/columns/DateTimeColumn.java (hunk 4 above)
Comment (IKnewOne) [reply lib/src/main/java/adminlte/entity_list_table/communication/http/tables/columns/DateTimeColumn.java:72]: Yeah. Not critical though ?

File: lib/src/main/java/adminlte/entity_list_table/communication/http/tables/AbstractTable.java (hunk 1 above)
Comment (IKnewOne) [reply lib/src/main/java/adminlte/entity_list_table/communication/http/tables/AbstractTable.java:48]: Made current table's columns as default

File: lib/src/main/java/adminlte/entity_list_table/communication/http/tables/AbstractTable.java (hunk 2 above)
Comment (IKnewOne) [reply lib/src/main/java/adminlte/entity_list_table/communication/http/tables/AbstractTable.java:50]: Expanded upon this in next commit

**Pull request merged**